service-orchestration/tests/tests.bal

```diff
@@ -0,0 +1,233 @@
+import ballerina/http;
+import ballerina/random;
+import ballerina/test;
+
+final http:Client cl = check new (string `http://localhost:${port}/healthcare/categories`);
+
+@test:Config
+function testSuccessfulRequest() returns error? {
+    ReservationStatus status = check cl->/surgery/reserve.post({
+        "patient": {
+            "name": "John Doe",
+            "dob": "1940-03-19",
+            "ssn": "234-23-525",
+            "address": "California",
+            "phone": "8770586755",
+            "email": "johndoe@gmail.com",
+            "cardNo": "7844481124110331"
+        },
+        "doctor": "thomas collins",
+        "hospital_id": "grandoaks",
+        "hospital": "grand oak community hospital",
+        "appointment_date": "2023-10-02"
+    });
+
+    final Appointment & readonly app;
+    lock {
+        app = <Appointment & readonly> appointment;
+    }
+    test:assertEquals(status, <ReservationStatus> {
+        "appointmentNo": app.appointmentNumber,
+        "doctorName": app.doctor.name,
+        "patient": app.patient.name,
+        "actualFee": app.doctor.fee,
+        "discount": 20,
+        "discounted": 5600.0,
+        "paymentID": "f130e2ed-a34e-4434-9b40-6a0a8054ee6b",
+        "status": "settled"
+    });
+}
+
+@test:Config
+function testUnknownCaregory() returns error? {
+    ReservationStatus|http:ClientError status = cl->/rheumatology/reserve.post({
+        "patient": {
+            "name": "John Doe",
+            "dob": "1940-03-19",
+            "ssn": "234-23-525",
+            "address": "California",
+            "phone": "8770586755",
+            "email": "johndoe@gmail.com",
+            "cardNo": "7844481124110331"
+        },
+        "doctor": "thomas collins",
+        "hospital_id": "grandoaks",
+        "hospital": "grand oak community hospital",
+        "appointment_date": "2023-10-02"
+    });
+
+    if status !is http:ClientRequestError {
+        test:assertFail("expected an http:ClientRequestError, found " + (typeof status).toString());
+    }
+
+    test:assertEquals(status.message(), "Not Found");
+    var detail = status.detail();
+    test:assertEquals(detail.statusCode, http:STATUS_NOT_FOUND);
+    test:assertEquals(detail.body, "unknown hospital, doctor, or category");
+}
+
+isolated Appointment? & readonly appointment = ();
+
+public client class MockHttpClient {
+    private final string url;
+
+    isolated function init(string url) {
+        self.url = url;
+    }
+
+    isolated resource function post [http:PathParamType... path](http:RequestMessage message, map<string|string[]>? headers = (), string?
```

Comment:
I feel like the line is too lengthy (138 characters). Should we stick with some best practices for line length?

Reply:
```
resource function post [http:PathParamType... path](http:RequestMessage message,
            map<string|string[]>? headers = (),
            string? mediaType = (),
            http:TargetType targetType = http:Response,
            *http:QueryParams params) returns http:Response|anydata|http:ClientError {
```